subsys/bluetooth/host/hci_core.c

```diff
@@ -4694,6 +4698,92 @@ int bt_enable(bt_ready_cb_t cb)
 	return 0;
 }
 
+static int set_ad(u16_t hci_op, const struct bt_data *ad, size_t ad_len)
+{
+	struct bt_hci_cp_le_set_adv_data *set_data;
+	struct net_buf *buf;
+	int i;
+
+	buf = bt_hci_cmd_create(hci_op, sizeof(*set_data));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	set_data = net_buf_add(buf, sizeof(*set_data));
+
+	memset(set_data, 0, sizeof(*set_data));
+
+	for (i = 0; i < ad_len; i++) {
+		/* Check if ad fit in the remaining buffer */
+		if (set_data->len + ad[i].data_len + 2 > 31) {
+			net_buf_unref(buf);
+			return -EINVAL;
+		}
+
+		set_data->data[set_data->len++] = ad[i].data_len + 1;
+		set_data->data[set_data->len++] = ad[i].type;
+
+		memcpy(&set_data->data[set_data->len], ad[i].data,
+		       ad[i].data_len);
+		set_data->len += ad[i].data_len;
+	}
+
+	return bt_hci_cmd_send_sync(hci_op, buf, NULL);
+}
+
+int bt_set_name(const char *name)
+{
+#if CONFIG_BT_DEVICE_NAME_STORAGE > 0
+	size_t len = strlen(name);
+	int err;
+
+	if (len > sizeof(bt_dev.name)) {
+		return -EINVAL;
+	}
+
+	if (!strcmp(bt_dev.name, name)) {
+		return 0;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		err = settings_save_one("bt/name", CONFIG_BT_DEVICE_NAME);
+		if (err) {
+			return err;
+		}
+	}
+
+	memset(bt_dev.name, 0, sizeof(bt_dev.name));
+	strncpy(bt_dev.name, name, sizeof(bt_dev.name) - 1);
```

Comment:
If you fix the length check I commented on earlier, I think the memset may become unnecessary, and you should then remove the `- 1` from the above line.

Reply:
Yep.